open-api/rest-catalog-open-api.yaml

```diff
@@ -541,6 +541,216 @@ paths:
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan/{id}:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+      - $ref: '#/components/parameters/id'
+
+    get:
+      tags:
+        - Catalog API
+      summary: Uses a `plan-id` as input to get status of `planTable`. Returns a list of `PlanTask`, a list of `FileScanTask`, or both when plan is complete.
+      operationId: GetTaskStatus
+      description:
+        Gets the status of a plan by using a `plan-id` which can be obtained from `planTable`.
+        If the plan is not completed, returns a `plan-status` representing the state of the plan.
+        If the plan is completed, returns a list of `PlanTask` or `FileScanTask`.
+      
+        If an invalid `plan-id` is provided, the service will return a `404 NoSuchPlanIdError` exception indicating that the requested plan does not exist.
+      responses:
+        200:
+          $ref: '#/components/responses/GetTasksStatusResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
+          description:
+            Not Found
+            - NoSuchPlanIdException, the plan-id does not exist
+            - NoSuchTableException, the table does not exist
+            - NoSuchNamespaceException, the namespace does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergErrorResponse'
+              examples:
+                PlanIdDoesNotExist:
+                  $ref: '#/components/examples/NoSuchPlanIdError'
+                TableDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        419:
+          $ref: '#/components/responses/AuthenticationTimeoutResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+    delete:
+      tags:
+        - Catalog API
+      summary: Uses a `plan-id` as input to cancel a `planTable` operation.
+      operationId: CancelPlan
+      description:
+        Uses a `plan-id` as input to cancel a `planTable` operation.
+        If successful, should return a state that says "cancelled".
+        
+        If an invalid `plan-id` is provided, the service will return a `404 NoSuchPlanIdError` exception indicating that the requested plan does not exist.
```

Comment:
I think there are other failure cases for this API where there's an invalid state change for the current plan task ->
"Cancelled -> Cancelled"
"Failed -> Cancelled"

There should probably be some 4xx error for this. Maybe the first one doesn't necessarily need to be a failure case either but definitely the second one (It doesn't make sense to cancel a failed plan).

Reply:
I addressed a few of these in my PR while reviewing. I think this should actually never return a 404. There's no need to cause an error on the client when this is an informational request to free up resources on the service. If the service doesn't know about the plan-id, then that's okay and it can ignore the request.

Reply:
Yeah good point, if I think about it for this specific case of cancellation if it were to fail, a client can't take any meaningful action since it's all server side processing for the cancellation. Resolving.